src/Automation/OutputFileHelper.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Axe.Windows.Abstractions;
+using System;
+using Path = System.IO.Path;
+
+namespace Axe.Windows.Automation
+{
+    internal class OutputFileHelper : IOutputFileHelper
+    {
+        private readonly string _outputDirectory;
+        private readonly ISystemDateTime _dateTime;
+
+        public const string DefaultOutputDirectoryName = "AxeWindowsOutputFiles";
+        public const string DefaultFileNameBase = "AxeWindows";
+
+        public OutputFileHelper(string outputDirectory, ISystemFactory systemFactory)
+        {
+            if (systemFactory == null) throw new ArgumentNullException(nameof(systemFactory));
+
+            _dateTime = systemFactory.CreateSystemDateTime();
+            if (_dateTime == null) throw new InvalidOperationException($"Expected {nameof(_dateTime)} not to be null");
```

Comment:
>_dateTime == null [](start = 16, length = 17)

Are there scenarios where the production code will return null for _dateTime, environment, or directory? If not, it seems like these checks are not helpful in production code